framework/include/problems/FEProblemBase.h

```diff
@@ -2047,10 +2047,21 @@ class FEProblemBase : public SubProblem, public Restartable
    */
   unsigned int nlSysNum(const NonlinearSystemName & nl_sys_name) const;
 
+  /**
+   * Whether it will skip further residual evaluations and fail the next nonlinear convergence check
+   */
+  bool getFailNextNonlinearConvergenceCheck() const
+  {
+    return _fail_next_nonlinear_convergence_check;
+  }
+
   /**
    * Skip further residual evaluations and fail the next nonlinear convergence check
    */
-  bool failNextNonlinearConvergenceCheck() const { return _fail_next_nonlinear_convergence_check; }
+  void setFailNextNonlinearConvergenceCheck(bool fail)
+  {
+    _fail_next_nonlinear_convergence_check = fail;
```

Comment:
Will this ever be called with a `false` argument? If not, I'd remove the argument

Reply:
that s what I expected the `failNextNonlinearConvergenceCheck` to do, which lost me half an hour or so

the current configuration is the most explicit

Reply:
Unfailing the solve. Heroic recovery

my concern now is what is people think that fail(false) means it will always not fail

Reply:
changed